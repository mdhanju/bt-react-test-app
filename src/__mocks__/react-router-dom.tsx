/* eslint-disable */

import React, { ReactNode } from 'react';

type Props = {
    children: ReactNode;
};

const rrd = require('react-router-dom');
// Just render plain div with its children
rrd.BrowserRouter = ({ children }: Props): ReactNode => <div>{children}</div>;
module.exports = rrd;
