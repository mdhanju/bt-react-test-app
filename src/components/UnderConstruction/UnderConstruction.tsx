import React from 'react'

import ContentHolder from '../ContentHolder'

const LoadingSpinner = (): React.ReactElement => {
    return <ContentHolder>Under Construction</ContentHolder>
}

export default LoadingSpinner
