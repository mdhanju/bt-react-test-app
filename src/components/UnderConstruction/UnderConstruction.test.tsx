import React from 'react'
import UnderConstruction from './UnderConstruction'
import { MemoryRouter } from 'react-router-dom'

describe('<UnderConstruction />', () => {
    it('should render properly without crashing', () => {
        const wrapper = shallow(<UnderConstruction />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('render the UnderConstruction', () => {
        const wrapper = render(
            <MemoryRouter>
                <UnderConstruction />
            </MemoryRouter>
        )
        expect(wrapper.html()).toMatch(
            '<div class="my-auto col">Under Construction</div>'
        )
    })
})
