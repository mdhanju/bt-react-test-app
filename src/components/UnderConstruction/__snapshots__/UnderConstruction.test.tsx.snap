// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnderConstruction /> should render properly without crashing 1`] = `
Array [
  <ContentHolder>
    Under Construction
  </ContentHolder>,
]
`;
